docs/NATIVE-SEGWIT-UPGRADE.md

```diff
@@ -0,0 +1,112 @@
+# UPGRADING YOUR JOINMARKET TO NATIVE SEGWIT (bech32)
+
+## Motivation: why?
+
+I guess the large majority of Joinmarket users know, but for completeness: native segwit p2wpkh is substantially (perhaps 25% on average?) cheaper than p2sh-p2wpkh which we currently used, and is rapidly becoming industry standard amongst most users wallets and other Bitcoin transaction operators, including other coinjoin systems, e.g. Wasabi - this latter point is very important in that it helps with privacy to join in with the crowd. In particular, Payjoins require both wallets to use the same scriptPubkey (or address) type.
+## Do I need to do anything at all?
+
+Technically, no, but realistically, yes.
+
+If you do nothing here's what will happen:
+
+### As a maker
+
+You can continue to run your maker bot (yield generator), serving Taker customers still using p2sh ('3') addresses with your existing wallet. The traffic on these will go down over time because (a) native segwit transactions are cheaper, and Takers bear most of that cost, and (b) the default new wallet type is now native segwit, so new entrants will be creating wallets of that type, and using the native segwit pit, and so ignoring your bots.
+
+### As a taker
+
+If you have an existing wallet and still want to do joins with it with '3' addresses, you can do so, but you must set `native=false` in `[POLICY]` section of `joinmarket.cfg` - assuming you don't have this setting already (the point here is that the *default* value of that setting has changed from `false` to `true` as of 0.8.0). If you are just using `wallet-tool` or direct-sending payments, you don't even need to make that change, but you may as well do it, for as long as you are using the old wallet type.
+
```

Comment:
Maybe add a note that it makes sense to not migrate completely to the new native segwit wallet immediately, but run two yield generator bots in parallel for some time, if user regularly mixes maker and taker roles (runs yield generator, but from time to time stops it and do sendpayment) and sends to p2sh-p2wpkh recipients using coinjoins?

Reply:
Would like to clarify here if the `native=true` or `false` setting only affects the type of wallet being generated or doing coinjoins with '3' or 'bc1' types? 

Is the orderbook the Taker chooses from determined by the Taker's wallet type or the `native=` setting?

Reply:
@openoms  thanks for raising this, you may have saved a rather stupid mistake from going through:

after a test on regtest it looks like I was wrong, in one specific but *very* important way: if a user uses `native=true` but tries to do a taker side coinjoin with a p2sh wallet, it seems that a mixed address coinjoin results - I had been sure that this would not work (i.e. that it would complain about the authorizing address being the wrong type), oops ... this will be very easy to fix but obviously we cannot release like that, so i will patch that before merging this and releasing.

It's still possible I'm wrong but I will check again carefully.

Reply:
> Maybe add a note that it makes sense to not migrate completely to the new native segwit wallet immediately, but run two yield generator bots in parallel for some time, if user regularly mixes maker and taker roles (runs yield generator, but from time to time stops it and do sendpayment) and sends to p2sh-p2wpkh recipients using coinjoins?

I don't want to do this really; it has always been the case that JM acting as a single-scriptpubkey-type "consensus" system, it causes a bit of a problem for anyone wanting to send directly out of the coinjoins into a different scriptpubkey type. This isn't changing, but just like when we went legacy-segwit, it's hardly surprising that a combination of inertia and this spk-stickiness means not everyone migrates all at once.
For people just directly/neutrally being JM wallet users, with no specific activity pattern that fixes them to one spk type, they should and will want to switch for the reasons mentioned. I don't want to complicate this section more than say "we recommend you upgrade but the old version is still there".